tidy3d/components/medium.py

```diff
@@ -2486,6 +2615,37 @@ def loss_upper_bound(self) -> float:
         """Not implemented yet."""
         raise SetupError("To be implemented.")
 
+    def _sel_custom_data_inside(self, bounds: Bound):
+        """Return a new custom medium that contains the minimal amount data necessary to cover
+        a spatial region defined by ``bounds``.
+
+
+        Parameters
+        ----------
+        bounds : Tuple[float, float, float], Tuple[float, float float]
+            Min and max bounds packaged as ``(minx, miny, minz), (maxx, maxy, maxz)``.
+
+        Returns
+        -------
+        CustomPoleResidue
+            CustomPoleResidue with reduced data.
+        """
+        if not self.eps_inf.does_cover(bounds=bounds):
+            log.warning("Eps inf spatial data array does not fully cover the requested region.")
+        eps_inf_reduced = self.eps_inf.sel_inside(bounds=bounds)
+        poles_reduced = []
+        for pole, residue in self.poles:
+
+            if not pole.does_cover(bounds=bounds):
+                log.warning("Pole spatial data array does not fully cover the requested region.")
```

Comment:
I wonder if it is necessary to warn at all. It is expected that some structure will be outside the requested region. It might raise tons of warning from a dispersive custom medium.

Reply:
actually, here `bounds` would represent intersection of region from `.subsection(region=region, ...)` and the bounding box of underlying structure or simulation itself (if background medium). So, if the original structure was put together correctly (that is, custom medium either covers it entirely or only one point is provided along a given dimension for constant extrapolation), then it won't produce any warnings